### ISSUE TITLE
only plot soma if it is actually defined

### ISSUE THREAD

**Issue by jefferis** [jefferis/rcatmaid#62]
- and use the radius?

**Comment by jefferis** [comment]
Using the radius is a nice idea, but we will see that this is not currently being transformed by xform_brain and friends. Maybe we need to think about scaling the radius.

If we do not use the internal radius, how do we signal that we want a specific radius. Presumably still by saying

```plot3d(n, soma=2)```

but that will result in all cell bodies being plotted. So maybe we still need to do the soma tag existence check on catmaid neurons and only plot valid somata. But there could still be times where it's handy to insist on having spheres plotted at the root/somata for all neurons. So how do we signal that? Perhaps one option is a negative number for the radius.

**Comment by jefferis** [comment]
@alexanderbates what would you feel are

1. the most helpful default behaviours for plotting cell bodies
2. whether we should be able to **insist** on plotting cell bodies even if they are just the root of the neuron rather than a real soma tagged as such

Thanks!

**Comment by jefferis** [comment]
Alex wrote as follows:

> If a soma is marked in Catmaid, then the default should be to plot it in r I guess. I think it is useful to be able to insist on plotting the cell body only rarely, when the primary neurite has been traced and not finished. However, most people that find the primary neurite do go on and find the cell body right away. If people use the plotting function incorrectly they might think a soma exists when and where it does not. By making a user unable to insist, it might also motivate them to finish the somata/highlight that there is a problem with the rooting of the skeleton? (btw, As far as I know, people don't take the radius attribute in Catmaid too seriously once a soma is found.)
> 
> (Maybe for some cases, being able to quickly highlight the root node might be useful, if the user was using some algorithm that used it for example, in the event that there is no soma)

**Comment by jefferis** [comment]
Possible approach going forwards

```
soma=TRUE always plot with default radius
soma=FALSE never plot
soma=NA plot if unambiguously present with default radius (what should this be? should it be different for catmaidneurons)
soma=-2 plot with radius 2 (if present)
soma=2 always plot with radius 2
```

Otherwise we may need to introduce a somarad argument to allow separate specification of a default radius (or the encoded one)

**Comment by jefferis** [comment]
I think for the time being, we should at least modify plot3d.catmaidneuron so that it only plots "real" somata.

**Comment by jefferis** [comment]
Have implemented something on https://github.com/jefferis/rcatmaid/tree/feature/plot3d-somapos that uses radius of soma if it is sensible and otherwise defaults back to whatever the default is for a neuron (2 currently).

Can try it out with:

```r
devtools::install_github("jefferis/rcatmaid@feature/plot3d-somapos")
```

@alexanderbates do you want to try?

**Comment by jefferis** [comment]
Closed by 0a2a667
